components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLECPProviderServlet.java

```
@@ -0,0 +1,115 @@
+        out.print(soapFault);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        try {
+            handleRequest(req, resp);
+        } finally {
+            SAMLSSOUtil.removeSaaSApplicationThreaLocal();
+            SAMLSSOUtil.removeUserTenantDomainThreaLocal();
+            SAMLSSOUtil.removeTenantDomainFromThreadLocal();
+        }
+    }
+
+
+    private void handleRequest(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        try {
+            if (req.getHeader("Authorization") == null) {
+                String err = "Authorization Header cannot be Empty";
+                log.error(err);
+                throw new IdentitySAML2ECPException(err);
+            }
+        MessageFactory messageFactory = MessageFactory.newInstance();
+        InputStream inStream = req.getInputStream();
+        SOAPMessage soapMessage = messageFactory.createMessage(new MimeHeaders(), inStream);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        soapMessage.writeTo(out);
+        String strMsg = new String(out.toByteArray(), Charset.forName("UTF-8"));
+
+        log.info("ECP Request : " + strMsg);
+        String samlRequest = SAMLSOAPUtils.decodeSOAPMessage(soapMessage);
+        SamlSSORequestWrapper samlSSORequestWrapper = new SamlSSORequestWrapper(req);
+        samlSSORequestWrapper.setParameter(SAMLSSOConstants.SAML_REQUEST, samlRequest);
+        samlSSORequestWrapper.setParameter(SAMLSSOConstants.ECP_ENABLED, "true");
+        RequestDispatcher dispatcher = req.getRequestDispatcher("/samlsso");
+
+        log.info("Forwarding the ECP request to SAMLSSO Servlet");
+        dispatcher.forward(samlSSORequestWrapper, resp);
+
+        } catch (IdentitySAML2ECPException e) {
+            SAMLSOAPUtils.sendSOAPFault(resp, e.getMessage(), SAMLSOAPUtils.SOAP_FAULT_CODE_CLIENT);
```

We are swallowing exception here. It neither prints in sendSOAPFault

---

> We are swallowing exception here. It neither prints in sendSOAPFault

fixed.